dht/token.hh

```diff
@@ -229,7 +231,9 @@ inline std::strong_ordering operator<=>(const token& t1, const token& t2) {
         return std::strong_ordering::greater;
     }
 }
-inline bool operator==(const token& t1, const token& t2) { return t1 <=> t2 == 0; }
+inline bool operator==(const token& t1, const token& t2) {
+    return t1._data == t2._data && t1._kind == t2._kind;
+}
```

Comment:
True, but I wanted to compare `data` first since for different keys we don't need to test kind at all.

Reply:
Need to check the x86 assembly.
It's possible that the default implementation can be optimized (e.g by using vector instructions)

Reply:
The compiler is going to do this branchlessly anyway, so the order doesn't matter.

And even if it did generate a branch, you shouldn't rely on the source code order for generating a particular order in the executable. `[[likely]]` would be probably more appropriate tool for that than shifting the source around.

Reply:
I don't think the compiler will make it branchless.  We can trick it with !((t1._data - t2._data) | (t1._kind - t2._kind)).

Intel APX introduces conditional compare instructions, so in about 20-25 years it will be branchless.